frontend/src/components/ViewTrips/trip.js

```diff
@@ -8,35 +14,124 @@ import React from 'react';
  *    and values for a Trip document.
  * @return Title of the trip (if it exists).
  */
-function createTitleElement(tripObj) {
+function getTitle(tripObj) {
   try {
-    if('name' in tripObj) {
+    if (DB.TRIPS_NAME in tripObj) {
       return tripObj.name;
     }
-    throw new Error(`Property 'name' is not defined in 'tripObj.'`);
+    throw new Error(`Property '${DB.TRIPS_NAME}' is not defined in 'tripObj.'`);
   } catch (error) {
     console.log(`Error in fetching trip title: ${error}`);
     return 'Unable to fetch trip title';
   }
 }
 
 /**
- * Component corresponding to the container containing an individual trip.
+ * Returns the description of the trip associated with the Trip document data
+ * `tripObj`.
  *
- * Temporarily, only the title and associated document ID are included in a
- * text element for each trip <div> element. This is done to simple test the
- * query functionality.
+ * @param {firebase.firestore.DocumentData} tripObj Object containing the fields
+ *    and values for a Trip document.
+ * @return Description of the trip (if it exists).
+ */
+function getDescription(tripObj) {
+  try {
+    if (DB.TRIPS_DESCRIPTION in tripObj) {
+      return tripObj.description;
+    }
+    throw new Error(
+        `Property '${DB.TRIPS_DESCRIPTION}' is not defined in 'tripObj.'`);
+  } catch (error) {
+    console.log(`Error in fetching trip description: ${error}`);
+    return 'Unable to fetch trip description';
+  }
+}
+
+/**
+ * Returns the date range of the trip associated with the Trip document data
+ * `tripObj`.
  *
- * TODO(Issue 17): Feed all the Trip Doc data to the UI.
+ * @param {firebase.firestore.DocumentData} tripObj Object containing the fields
+ *    and values for a Trip document.
+ * @return Date range of the trip (if it exists).
+ */
+function getDateRange(tripObj) {
+  try {
+    if (DB.TRIPS_START_DATE in tripObj && DB.TRIPS_END_DATE in tripObj) {
+      const startDate = tripObj.start_date.toDate();
+      const endDate = tripObj.end_date.toDate();
+      return `${startDate.getMonth() + 1}/${startDate.getDate()}/`  +
```

Comment:
what kind of syntax is this?

Reply:
+1, have never seen this syntax either

Reply:
Do you mean the back-tick string syntax? They are called [template strings](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals).

If you meant something else, let me know.